pilot/pkg/model/service.go

```diff
@@ -481,8 +492,16 @@ type IstioEndpoint struct {
 	// Labels points to the workload or deployment labels.
 	Labels labels.Instance
 
-	// Address is the address of the endpoint, using envoy proto.
-	Address string
+	// Addresses is the address of the endpoint, using envoy proto:
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/endpoint/v3/endpoint_components.proto#config-endpoint-v3-endpoint-additionaladdress
+	// This change can support multiple addresses for an endpoint, then there are 2 use cases for it, such as
+	// 1. An endpoint can have both ipv4 or ipv6.
+	// 2. An endpoint can be represented a serviceentry/workload instance with multiple IP addresses.
```

Comment:
This case doesn't make sense to me. Can you show an example of how you can set this up?

AFAIK this would result in multiple Endpoints, not 1 endpoint with multiple addresses